person.rb

```diff
@@ -0,0 +1,25 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, parent_permission: true, name: 'Unknown')
+    @id = generate_random_id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def generate_random_id
+    rand(1..100)
+  end
```

Comment:
- [OPTIONAL] I think it could be better instead to create a method that generates a random id and then call the method in initialize you can directly do it in initialize, it is just one line 👉🏾 `@id = Random.rand(1..100)` 👍🏾